core/src/saros/session/User.java

```diff
@@ -54,7 +55,74 @@
 
   private final int favoriteColorID;
 
+  @Deprecated
   private volatile Permission permission = Permission.WRITE_ACCESS;
+  
+  
+  
+  
+  
+  
+  /* More flexible Permissions also known as Privilege */
+  private Map<UserPrivilege.Keys, UserPrivilege> privileges;
+  public Map<UserPrivilege.Keys, UserPrivilege> getPrivileges() {
+      return this.privileges;
```

Comment:
Map can be mutated by the caller.

Reply:
I do not understand how a get method may mutate the instance? Do you think privileges will mutate?